lib/matplotlib/tests/test_legend.py

```diff
@@ -508,6 +508,39 @@ def test_shadow_framealpha():
     assert leg.get_frame().get_alpha() == 1
 
 
+@image_comparison(['shadow_color.svg'])
+def test_shadow_color():
+    # Test whether a shadow of the appropriate color
+    # is generated when shadow's value is colorlike
+    fig, ax = plt.subplots()
+    ax.plot(range(100), label="test")
+    legends = {
+        'true': plt.legend(loc="upper left", shadow=True),
+        'false': plt.legend(loc="center left", shadow=False),
+        'default': plt.legend(loc="lower left"),
+        'colorstring': plt.legend(loc="upper right", shadow='red'),
+        'colortuple': plt.legend(loc="center right", shadow=(0.1, 0.2, 0.5)),
+        'colortab': plt.legend(loc="lower right", shadow='tab:cyan')
+    }
+    for x in legends:
+        plt.gca().add_artist(legends[x])
```

Comment:
I don't understand this; doesn't `plt.legend` add the legend to the plot already? Also, use `ax.legend` instead.